### DISCUSSION TITLE
Add custom visitor to display evaluated expression

### DISCUSSION THREAD

**Discussion by Bykiev** [ncalc/ncalc#354] [Ideas]
There are multiple issues #39 and #48 for this feature. Is it a good idea to implement it in NCalc or should we close this issue and users should implement it by self?

@gumbarros, what do you think?

**Comment by gumbarros** [comment]
As this is useful for debug, don't introduce any external dependency and many users want this, I see no problem in creating a method that execute the visitor and returns the string. I just don't know a good name for the method and visitor. Probably something like `ParameterizedSerializationVisitor`

**Reply by Bykiev** [reply]
Sounds good
